x-pack/platform/packages/shared/file-upload-common/src/telemetry/register_telemetry.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AnalyticsServiceSetup } from '@kbn/core/public';
+
+/**
+ * Event types.
+ */
+export const FILE_UPLOAD_EVENT = {
+  FILE_ANALYSIS: 'file_upload.file_analysis',
+  FILE_UPLOAD: 'file_upload.file_upload',
+  UPLOAD_SESSION: 'file_upload.upload_session',
+};
+
+/**
+ * Registers the file upload analytics events.
+ */
+export const registerFileUploadAnalyticsEvents = (analytics: AnalyticsServiceSetup) => {
+  analytics.registerEventType({
+    eventType: FILE_UPLOAD_EVENT.FILE_ANALYSIS,
+    schema: {
+      upload_session_id: {
+        type: 'keyword',
+        _meta: {
+          description: 'The unique identifier for the upload session.',
+        },
+      },
+      file_id: {
+        type: 'keyword',
+        _meta: {
+          description: 'The unique identifier for the file being uploaded.',
+        },
+      },
+      location: {
```

Comment:
Are we able to track files uploaded in Maps too?

Reply:
Created https://github.com/elastic/kibana/issues/241786 for adding telemetry for uploads from Maps.